src/ngx_http_lua_module.c

```diff
@@ -673,6 +701,10 @@ ngx_http_lua_init_main_conf(ngx_conf_t *cf, void *conf)
         lmcf->max_running_timers = 256;
     }
 
+    if (lmcf->semaphore_mm->num_per_block == NGX_CONF_UNSET_UINT) {
+        lmcf->semaphore_mm->num_per_block = 4094;
```

Comment:
Maybe add a comment explaining the magical number 4094 a bit? It's not a power of 2 and deserves a little explanation I suppose :)